### PR TITLE
EDU-751: Add caution blurb to StartToCloseTimeout

### PR DESCRIPTION
**Pull request by taonic**
## What does this PR do?

Adding a caution section to StartToCloseTimeout after issue reported by customer and a follow up discussion with @mfateev  and @cretz.

image

### CODE REVIEW

**Comment by cretz** [comment]
For this use case of worker crashing it could have been solved with start-to-close or heartbeat timeout. See the internal Slack thread on SDK channel with @mfateev :+1:

Having said that, I support only talking about start-to-close and not even mentioning heartbeat timeout if we want.

File: docs/concepts/activities.md
Comment (djmagee) [inline docs/concepts/activities.md:207]: **Please move these changes to `docs-src/concepts/what-is-a-start-to-close-timeout.md` so that they aren't lost when `docs/concepts/activities.md` is generated.**

The following edits conform to our [style guide](https://github.com/temporalio/documentation/blob/main/STYLE.md) and aren't meant to preclude the technical issues raised by @tsurdilo or @cretz.

```suggestion
We strongly recommend setting Start-To-Close Timeout or Heartbeat Timeout.
The Temporal Server does not detect Worker crashes directly and relies on these timeouts to force Activity retries.
```

**Review by djmagee — CHANGES_REQUESTED** [review]

File: docs/concepts/activities.md
Comment (taonic) [reply docs/concepts/activities.md]: Thanks @djmagee. I don't know why I thought there is no docs-src for this page.. Updated.

**Comment by taonic** [comment]
@tsurdilo @cretz it's ready for another look 🙏

File: docs-src/concepts/what-is-a-start-to-close-timeout.md
Comment (djmagee) [inline docs-src/concepts/what-is-a-start-to-close-timeout.md:23]: Suggest changing "take place" to "complete".

File: docs-src/concepts/what-is-a-start-to-close-timeout.md
Comment (djmagee) [inline docs-src/concepts/what-is-a-start-to-close-timeout.md:30]:
```suggestion
The Temporal Server does not detect failures when a Worker loses communication with the Server or crashes.
Therefore, the Temporal Server relies on this timeout to force Activity retries.
```

**Review by cretz — APPROVED** [review]
LGTM, but wait for @tsurdilo's approval

**Review by djmagee — APPROVED** [review]

File: docs-src/concepts/what-is-a-start-to-close-timeout.md
Comment (flossypurse) [inline docs-src/concepts/what-is-a-start-to-close-timeout.md:26]: Caution is so scary...

Can easily make this a happy `info` or `success` admonition.
We want it to be a good thing :)

**Review by flossypurse — APPROVED** [review]

**Pull request merged**
